doc/userguide/src/ExtendingRobotFramework/CreatingTestLibraries.rst

```diff
@@ -1943,6 +1943,63 @@ __ `Specifying argument types using @keyword decorator`_
 
 .. note:: Automatic type conversion is new in Robot Framework 3.1.
 
+Asynchronous Keywords
+~~~~~~~~~~~~~~~~~~~~~
+
+Starting from Robot Framework 6.1, it is possible to run async functions
+from the test libraries just like normal functions. For example:
+
+.. sourcecode:: python
+
+    import asyncio
+    from robot.api.deco import keyword
+
+
+    @keyword
+    async def this_keyword_waits():
+        await asyncio.sleep(5)
+
+Running an async function inside asyncio.run() works, but it's not recommended
+as this will close the event loop as soon as the function finishes. You can also
+get the reference of the loop by running asyncio.get_running_loop(), note however
+that you should be careful when modifying how the event loop runs, for example
+calling close() on it will make it impossible to run other coroutines, let Robot
+Framework manage the loop. If you have any function or resource that requires the
+event loop, even though await is not used, you have to define your function as
+async to have the event loop available.
```

Comment:
1. References to code should be surrounded with backticks like
    ```
    function inside `asyncio.run()` works
    ```

2. We have generally used style `it is` instead of `it's` in this somewhat formal text.